es-space-mgmt/main/default/lwc/customerDetailForm/__tests__/customerDetailForm.test.js

```diff
@@ -0,0 +1,143 @@
+
+        const RECORD_ID_INPUT = '0031700000pJRRSAA4';
+        const OBJECT_API_NAME_INPUT = 'Contact';
+        const RECORD_FIELDS_OUTPUT = [
+            'MailingCity',
+            'Email',
+            'Name',
+            'MailingState'
+        ];
+
+        // Create element
+        const element = createElement('c-customer-detail-form', {
+            is: CustomerDetailForm
+        });
+        element.sobjecttype = OBJECT_API_NAME_INPUT;
+        element.recordid = RECORD_ID_INPUT;
+        document.body.appendChild(element);
+
+        return flushPromises().then(() => {
+            // Validate if correct parameters have been passed to base components
+            const formEl = element.shadowRoot.querySelector(
+                'lightning-record-form'
+            );
+
+            expect(formEl.fields).toEqual(RECORD_FIELDS_OUTPUT);
+            expect(formEl.recordId).toBe(RECORD_ID_INPUT);
+            expect(formEl.objectApiName).toBe(OBJECT_API_NAME_INPUT);
+        });
+    });
+
+    it('handles form submission succesfully and fires a custom event', () => {
+        // Assign mock value for resolved Apex promise
+        getCustomerFields.mockResolvedValue(APEX_GET_CUSTOMER_FIELDS_SUCCESS);
+
+        const EVENT_DETAIL_PARAMETER = 'NY';
+
+        const RECORD_ID_INPUT = '0031700000pJRRSAA4';
+        const OBJECT_API_NAME_INPUT = 'Contact';
+
+        // Create element
+        const element = createElement('c-customer-detail-form', {
+            is: CustomerDetailForm
+        });
+        element.sobjecttype = OBJECT_API_NAME_INPUT;
+        element.recordid = RECORD_ID_INPUT;
+        document.body.appendChild(element);
+
+        // Mock handler for child event
+        const customerUpdateHandler = jest.fn();
+        // Add event listener to catch child event
+        element.addEventListener('customerupdate', customerUpdateHandler);
+
+        return flushPromises()
+            .then(() => {
+                // Validate if correct parameters have been passed to base components
+                const formEl = element.shadowRoot.querySelector(
+                    'lightning-record-form'
+                );
+                formEl.dispatchEvent(
+                    new CustomEvent('success', { detail: FORM_SUBMIT_RESPONSE })
+                );
+            })
+            .then(() => {
+                expect(customerUpdateHandler).toHaveBeenCalled();
+                expect(customerUpdateHandler.mock.calls[0][0].detail).toEqual(
+                    EVENT_DETAIL_PARAMETER
+                );
+            });
+    });
+
+    it('shows error panel element', () => {
+        // Assign mock value for rejected Apex promise
+        getCustomerFields.mockRejectedValue(APEX_GET_CUSTOMER_FIELDS_ERROR);
+
+        // Create element
+        const element = createElement('c-customer-detail-form', {
+            is: CustomerDetailForm
+        });
+        element.sobjecttype = 'Lead';
```

Comment:
You don't need to set these element properties since you're not using them in this test.

Reply:
These are required to render the component. If we don't pass these, the component will not render and neither will show an error.

Reply:
ok sorry disregard then